qiskit_nature/operators/second_quantization/fermionic_op.py

```diff
@@ -0,0 +1,304 @@
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""The Fermionic-particle Operator."""
+
+from numbers import Number
+from typing import cast, Dict, List, Tuple, Union
+
+import numpy as np
+from qiskit.opflow import PauliSumOp
+
+from qiskit_nature import QiskitNatureError
+
+from .particle_op import ParticleOp
+
+
+class FermionicOp(ParticleOp):
+    """
+    Fermionic type operators
+
+    The abstract fermionic registers are implemented in two subclasses, FermionicOperator and
+    FermionicOp, inspired by the implementation of Pauli operators in qiskit. A
+    FermionicOperator is the equivalent of a single Pauli string on a qubit register.
+    The FermionicOp represents a sum of multiple FermionicOperators. They act on fermionic
+    registers of a fixed length determined at the time of initialization.
+    """
+
+    def __init__(self, data, coeff=1, register_length=None):
+        if not isinstance(data, (tuple, list, str)):
+            raise QiskitNatureError("Invalid input data for FermionicOp.")
+
+        if isinstance(data, tuple):
+            if isinstance(data[0], str) and isinstance(data[1], Number):
+                label = data[0]
+                if not self._validate_label(label):
+                    raise QiskitNatureError(
+                        "Label must be a string consisting only of "
+                        f"['I','+','-','N','E'] not: {label}"
+                    )
+                coeff = data[1]
+                self._register_length = len(label)
+                self._labels = [label]
+                self._coeffs = [coeff]
+            else:
+                raise QiskitNatureError(
+                    "Data tuple must be (str, Number), "
+                    f"but ({type(data[0])}, {type(data[1])}) is given."
+                )
+
+        elif isinstance(data, str):
+            label = data
+            if not self._validate_label(label):
+                raise QiskitNatureError(
+                    "Label must be a string consisting only of "
+                    f"['I','+','-','N','E'] not: {label}"
+                )
+            self._register_length = len(label)
+            self._labels = [label]
+            self._coeffs = [coeff]
+
+        elif isinstance(data, list):
+            if not data and register_length:
+                self._labels = ["I" * register_length]
+                self._coeffs = [0]
+            elif not data:
+                raise QiskitNatureError(
+                    "Empty data requires register_length parameter."
+                )
+            else:
+                self._register_length = len(data[0][0])
```

Comment:
If you frequently create operators with the same coefficients, I would like to create this or similar.
How often do you expect this?

Reply:
I cannot really say how often this may occur but I think it is reasonable to assume that this occurs often enough to be supported. We can provide a limited version in which we allow either a 2D-list or a 1D-list with either a single coeff or a list of coeffs with matching length? I think this should be doable and maintainable with the least amount of edge cases.. :thinking:

Reply:
This is a quite common occurrence in the case of Lattice models (e.g. think of the hopping and on site interaction terms in the Fermi-Hubbard model). It is not so common in molecular problems.

Reply:
How about `0.25 * sum(FermionicOp(label) for label in ['NIII', 'INII', 'IINI', 'IIIN'])` for this?

Reply:
I am fine with an approach like this. But if we choose to do so we need to document this properly. I.e. we need to carefully explain the kind of list (i.e. a 2D list) which is required for the constructor and we should add an example like the one above to show how a user can get around it.